src/request.rs

```diff
@@ -277,6 +277,47 @@ impl RequestBuilder {
         Ok(self)
     }
 
+    /// Sends a multipart/formdata body.
+    ///
+    /// ```no_run
+    /// use reqwest::mime;
+    /// use reqwest::{MultipartRequest, MultipartField, to_multipart};
+    ///
+    /// let client = reqwest::Client::new().unwrap();
+    /// let response = client.post("your url").unwrap()
+    ///     .multipart(
+    ///         // Add fields from anything serializable
+    ///         to_multipart(&[("key", "value"), ("key2", "value2")]).unwrap()
+    ///         // Add fields builder style
+    ///         .field(MultipartField::param("key3", "value3"))
+    ///         .field(MultipartField::param("json", "{ \"number\": 5 }")
+    ///             .mime(Some(mime::APPLICATION_JSON)))
+    ///         .field(MultipartField::file("file", "/path/to/file")
+    ///             .expect("File could not be opened"))
+    ///     ).send();
+    ///
+    /// ```
+    ///
+    /// See [`to_multipart`](fn.to_multipart.html), [`MultipartRequest`](struct.MultipartRequest.html)
+    /// and [`MultipartField`](struct.MultipartField.html) for more examples.
+    pub fn multipart(&mut self, multipart: MultipartRequest) -> &mut RequestBuilder {
```

Comment:
Sorry, I've been busy the past few days trying to get the http crate ready for preview. I'm going to download this branch locally and try some things with the serializer, instead of asking you to spend your time trying out crazy ideas I spout out.